app/scripts/services/hpa.js

```diff
@@ -146,15 +168,27 @@ angular.module("openshiftConsole")
         return $q.when([]);
       }
       return MetricsService.isAvailable().then(function(metricsAvailable) {
+        var v2HPAWarning = hasV2HPAScaler(scaleTarget, hpaResources);
         return _.compact([
           hasMetricsAvailableWarning(metricsAvailable),
-          hasCPURequestWarning(scaleTarget, limitRanges, project),
+          v2HPAWarning ? v2HPAWarning : hasCPURequestWarning(scaleTarget, limitRanges, project),
```

Comment:
I think this would be easier to read as...

```js
hasMetricsAvailableWarning(metricsAvailable),
// Only check for CPU request if not using v2 HPA metrics.
hasV2HPAScaler(scaleTarget, hpaResources) || hasCPURequestWarning(scaleTarget, limitRanges, project),
hasCompetingAutoscalersWarning(hpaResources),
// ...
```